packages/augur-core/source/contracts/reporting/RepPriceOracle.sol

```diff
@@ -45,73 +67,82 @@ contract RepPriceOracle is IRepPriceOracle, Initializable {
-        if (_exchange == IUniswapV2(0)) {
-            _exchangeData.price = getInitialPrice(_reputationToken);
-            return _exchangeData;
+        if (_blockNumber != _exchange.blockNumberLast()) {
+            _exchange.sync();
         }
-        (uint256 _token0Amount, uint256 _token1Amount) = _exchange.getReservesCumulative();
-        if (_token0Amount == 0 || _token1Amount == 0) {
+        uint256 _repPriceCumulative = _exchangeData.repIsToken0 ? _exchange.price0CumulativeLast() : _exchange.price1CumulativeLast();
+        if (_repPriceCumulative == 0) {
             return _exchangeData;
         }
 
-        uint256 _cashAmountAccumulated = cash < address(_reputationToken) ? _token0Amount : _token1Amount;
-        uint256 _repAmountAccumulated = cash < address(_reputationToken) ? _token1Amount : _token0Amount;
-
-        uint256 _cashAccumulationDelta = _cashAmountAccumulated - _exchangeData.cashAmountAccumulated;
-        uint256 _repAccumulationDelta = _repAmountAccumulated - _exchangeData.repAmountAccumulated;
-        // TODO: This should not be possible normally. When the real Uniswap contracts are in this condition can be removed.
-        if (_repAccumulationDelta == 0) {
+        // The first time we have actual data from the exchange we need to simply record the acumulator value and return the default since we have no accurate _relative_ delta to base the real price off of
+        if (_exchangeData.repPriceAccumulated == 0) {
+            _exchangeData.blockNumber = _blockNumber;
+            _exchangeData.blockTimestamp = _blockTimestamp;
+            _exchangeData.repPriceAccumulated = _repPriceCumulative;
             return _exchangeData;
         }
-        uint256 _deltaPrice = _cashAccumulationDelta * 1 ether / _repAccumulationDelta;
 
-        uint256 _weight = getWeight(_blockNumber - _exchangeData.blockNumber);
-        uint256 _price = (((1 ether - _weight) * _exchangeData.price) + (_weight * _deltaPrice)) / 1 ether;
+        uint256 _blocksElapsed = _blockNumber - _exchangeData.blockNumber;
 
-        _exchangeData.price = _price;
-        _exchangeData.blockNumber = _blockNumber;
-        _exchangeData.cashAmountAccumulated = _cashAmountAccumulated;
-        _exchangeData.repAmountAccumulated = _repAmountAccumulated;
+        uint256 _price = (_repPriceCumulative - _exchangeData.repPriceAccumulated) * 10**18 / _blocksElapsed / Q112;
+        require(_price > 0, "Price should not be 0");
 
-        return _exchangeData;
-    }
+        uint256 _secondsElapsed = _blockTimestamp - _exchangeData.blockTimestamp;
+        uint256 _priceAverage = _price;
 
-    function getWeight(uint256 _blockDelta) public view returns (uint256) {
-        uint256 _weight = _blockDelta * 1 ether / tau;
-        if (_weight > 1 ether) {
-            return 1 ether;
+        if (_secondsElapsed < period) {
+            _priceAverage = (_exchangeData.price * (period - _secondsElapsed) + _price * _secondsElapsed) / period;
```

Comment:
Probably worth a comment indicating why safe math isn't necessary here (or anywhere else in this file).  Alternatively, use safe math.